slick/src/main/scala/slick/lifted/Query.scala

```diff
@@ -155,6 +155,11 @@ sealed abstract class Query[+E, U, C[_]] extends QueryBase[C[U]] { self =>
     new WrappingQuery[(G, Query[P, U, Seq]), (T, Query[P, U, Seq]), C](group, key.zip(value))
   }
 
+  /** Specify part of a select statement for update and marked for row level locking */
+  def forUpdate: Query[E, U, C] = {
```

Comment:
I'm not convinced about the design. Is `forUpdate` really a query language feature? For instance, what happens when you put it into a subquery? Does this have sensible semantics that are different from putting it at the top level (or nowhere)? I would assume that it's rather a feature of query *execution*. A natural place for this would be an overload or flag in `Query.result` (e.g. `q.result(locked = true)`) or a new method (`q.lockedResult`) that is only available in `SqlProfile`. The flag could be passed directly to the code generator (similar to how forced insert vs soft insert is handled), thus avoiding the changes to the AST and `mergeToComprehensions`.